### ISSUE TITLE
Carousel button links don't work with space keys

### ISSUE THREAD

**Issue by masi** [twbs/bootstrap#32824]
The example in the 5.0 docs uses a-tags with role=button. To fully implement the ARIA pattern, please either change the example to use button-tags or fix the JS code to invoke the action with a space key.

https://www.w3.org/TR/wai-aria-practices-1.1/#button
https://dev.to/tylerjdev/when-role-button-is-not-enough-dac

**Comment by knoobie** [comment]
There is already a PR https://github.com/twbs/bootstrap/pull/32627.

**Comment by ffoodd** [comment]
This stays accurate IMHO, since carousel will still allow to use links. @patrickhlauke do you agree, or could we skip this?

**Comment by patrickhlauke** [comment]
should ideally do a brutal "if `role="button"` apply handler for `Space`" (possibly just scoped to bootstrap's components)

**Comment by patrickhlauke** [comment]
(noting that it's not a WCAG failure per se, but yes the behaviour should be that things that are announced as buttons will trigger with both `Enter` and `Space`)

**Comment by masi** [comment]
> There is already a PR #32627.

Sorry, my bad. Didn't found that one in a quick search.

Thank you, @patrickhlauke

**Comment by patrickhlauke** [comment]
reminded of this https://patrickhlauke.github.io/aria/presentation/?full#103 - could possibly do a generic one that targets all known bootstrap components where authors may end up using links or generic containers with `role="button"` (dropdown toggles, carousel buttons/indicators, close buttons, collapse triggers, ...)

**Comment by patrickhlauke** [comment]
Closed via https://github.com/twbs/bootstrap/pull/32627
